### PR TITLE
Cross platform apps

### PR DESCRIPTION
**Pull request by nid90**
**Partially Closes:** #452 

## Because

People want to synchronize the release cycle of their cross-platform apps across both platforms – iOS and Android.

## This addresses

- Adds the ability to add cross-platform apps (under a feature flag)
- Changes the release train schema under the hood to accommodate the cross-platform synced release trains 
- Migrates all single-platform release trains to the above schema
- Directory structure cleanup for release-related models, controllers, and routes
- Persist changelog for releases in the db for posterity

### CODE REVIEW

File: db/migrate/20230606084424_add_train_groups_and_train_group_runs.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+class AddTrainGroupsAndTrainGroupRuns < ActiveRecord::Migration[7.0]
+  def change
+    create_table :train_groups, id: :uuid do |t|
+      t.belongs_to :app, null: false, foreign_key: true, type: :uuid
+
+      t.string :name, null: false
+      t.string :description, null: false
```
Comment (nid90) [inline db/migrate/20230606084424_add_train_groups_and_train_group_runs.rb:7]: `description` should be `null: true`

File: app/models/integration.rb [hunk 2]
```diff
@@ -135,6 +138,14 @@ def store_provider
       build_channel.find(&:store?)&.providable
     end
 
+    def android_store_provider
+      build_channel.find { |chan| chan.providable_type == "GooglePlayStoreIntegration" }&.providable
```
Comment (kitallis) [inline app/models/integration.rb:142]:
```suggestion
      build_channel.find(&:google_play_store_integration?)&.providable
```

File: app/models/integration.rb [hunk 3]
```diff
@@ -135,6 +138,14 @@ def store_provider
...
+    end
+
+    def ios_store_provider
+      build_channel.find { |chan| chan.providable_type == "AppStoreIntegration" }&.providable
```
Comment (kitallis) [inline app/models/integration.rb:146]:
```suggestion
      build_channel.find(&:app_store_integration?)&.providable
```

File: app/controllers/release_platforms_controller.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+class ReleasePlatformsController < SignedInApplicationController
+  using RefinedString
```
Comment (kitallis) [inline app/controllers/release_platforms_controller.rb:2]: Remove

**Review by kitallis — APPROVED** [review]
phew what a beast ❇️

**Pull request merged**